### PR TITLE
WT-9404 Review of transaction and timestamp documentation

### PR DESCRIPTION
**Pull request by agorrod**
@keithbostic and @sauclovian-wt I did a pass through the documentation and had a few questions/thoughts.

I figured this was probably the easiest way to start that conversation, so I pushed it up.

I didn't spend a lot of time on the changes - since I didn't want to waste anyone time if my understanding is wrong or out of date.. So please take a skim through and let me know if the suggested changes are on-track or off-track.

### CODE REVIEW

File: src/docs/transactions_api.dox [hunk 1]
```diff
@@ -151,7 +157,8 @@ than snapshot, or if the current transaction has performed any write operations.
 
 \warning
 Applications not using read timestamps for search may see different results
-after the snapshot is updated.
+after the snapshot is updated. NOTE: Is this true? It shouldn't be if the set
+of actively updated timestamps are disjoint from those being read.
```
Comment (keithbostic) [inline src/docs/transactions_api.dox:161]: If they aren't disjoint though? The sentence says "may", or am I missing your point?

Comment (agorrod) [reply src/docs/transactions_api.dox:161]: The question I was trying to ask here: Should we be more explicit about saying that the expected model is that writes and reads happen in disjoint ranges when using timestamps (i.e: there is no snapshot mechanism managing visibility).
Is that covered somewhere else?

**Review by agorrod — COMMENTED** [review]
Sorry for  the extended delay here.

I've got one remaining question (to which I think the answer is: nothing more to do).

Then let me know if you think this suggested change should be discarded or cleaned up and merged.

File: src/docs/transactions_api.dox (hunk 1 above)
Comment (keithbostic) [reply src/docs/transactions_api.dox:161]: I agree most applications would be looking at disjoint ranges, but I can imagine a trigger thread looking for rows that change, and that application would loop, alternating between checking some set of k/v pairs and resetting the snapshot. (That said, I'm struggling to imagine that thread setting a read timestamp at all.)

Anyway, I don't object to the idea and it's not covered anywhere else, I don't think. Do you have suggested wording?

How about something like:
```
Applications repeatedly reading the same rows, without specifying read timestamps for search, may see
different results after the snapshot is updated.
```

**Comment by agorrod** [comment]
I have done a pass to tidy this up. I don't think any of the changes are particularly critical, so take it or push back as you see fit.

**Review by keithbostic — APPROVED** [review]

**Pull request merged**